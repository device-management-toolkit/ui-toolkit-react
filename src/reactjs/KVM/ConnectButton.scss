/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/

.button{
    margin-right: 3px;
    padding: 2px;
    min-width: 90px
}