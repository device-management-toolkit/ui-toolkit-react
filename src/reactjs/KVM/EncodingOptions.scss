/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/
 
.encoding {
    margin-left: 10px;
  }
  .encoding label {      
    padding: 2px;
    font-size: 15px;
  }
  .encoding select {
    padding: 2px 0;
  }
  
  .reldisabled{
    opacity: 0.5;
  }