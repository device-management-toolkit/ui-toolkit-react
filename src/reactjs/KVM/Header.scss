/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/
 
.header {
  background-color: darkgray;
  padding: 5px;
  font-size: 13px;
}
