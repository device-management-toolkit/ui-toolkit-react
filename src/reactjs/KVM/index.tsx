/*********************************************************************
* Copyright (c) Intel Corporation 2019
* SPDX-License-Identifier: Apache-2.0
**********************************************************************/
import React from 'react'
import { render } from 'react-dom'
import { KVM } from './UI'
import i18n from '../../i18n'
// Get browser language
i18n.changeLanguage(navigator.language).catch(() => console.info('error occured'))

const url = new URL(window.location.href)
const params = new URLSearchParams(url.search)

render(<KVM autoConnect={false} deviceId={params.get('deviceId')} mpsServer={params.get('mpsServer') + '/relay'} authToken="authorization" mouseDebounceTime={200} canvasHeight={'100%'} canvasWidth={'100%'} />, document.querySelector('#kvm'))
