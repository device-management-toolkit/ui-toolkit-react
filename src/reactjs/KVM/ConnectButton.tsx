/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/

import React from 'react'
import './ConnectButton.scss'

export interface ConnectProps {
  kvmstate: number
  handleConnectClick: (e: any) => void
}

export class ConnectButton extends React.Component<ConnectProps, {}> {
  render (): React.ReactNode {
    return (
      <button className="button" onClick={this.props.handleConnectClick}>
        {this.props.kvmstate === 1 ? 'Connecting KVM' : (this.props.kvmstate === 2 ? 'Disconnect KVM' : 'Connect KVM')}
      </button>
    )
  }
}
