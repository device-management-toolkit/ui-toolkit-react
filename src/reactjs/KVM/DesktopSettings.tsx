/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/

import React from 'react'
import { EncodingOptions } from './EncodingOptions'

export interface IDesktopSettings {
  changeDesktopSettings: (settings: any) => void
  getConnectState: () => number
}

export class DesktopSettings extends React.Component<IDesktopSettings> {
  desktopsettings = {
    encoding: 1
  }

  constructor (props: IDesktopSettings) {
    super(props)
    this.changeEncoding = this.changeEncoding.bind(this)
  }

  changeEncoding (encoding: number): void {
    this.desktopsettings.encoding = encoding
    this.props.changeDesktopSettings(this.desktopsettings)
  }

  render (): React.ReactNode {
    return (
      <EncodingOptions changeEncoding={this.changeEncoding} getConnectState={this.props.getConnectState}/>
    )
  }
}
