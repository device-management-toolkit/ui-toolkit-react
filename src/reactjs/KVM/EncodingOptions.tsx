/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/

import React from 'react'
import './EncodingOptions.scss'

export interface IEncodingOptions {
  changeEncoding: (encoding: number) => void
  getConnectState: () => number
}

export class EncodingOptions extends React.Component<IEncodingOptions, { value: number }> {
  constructor (props: IEncodingOptions) {
    super(props)
    this.state = { value: 1 }
    this.onEncodingChange = this.onEncodingChange.bind(this)
  }

  onEncodingChange (e): void {
    // e.persist();
    this.setState({ value: e.target.value })
    this.props.changeEncoding(e.target.value)
  }

  render (): React.ReactNode {
    return (
      <span className="encoding">
        <label >Encoding:</label>
        <select value={this.state.value} className= {this.props.getConnectState() === 2 ? 'reldisabled' : ''} onChange={this.onEncodingChange} disabled={this.props.getConnectState() === 2}>
          <option value="1">RLE 8</option>
          <option value="2">RLE 16</option>
        </select>
      </span>
    )
  }
}
