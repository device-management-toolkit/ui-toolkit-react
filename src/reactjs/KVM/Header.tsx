/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/

import React from 'react'
import { ConnectButton } from './ConnectButton'
import { DesktopSettings } from './DesktopSettings'
import './Header.scss'

export interface IHeaderProps {
  kvmstate: number
  deviceId: string | null
  server: string | null
  handleConnectClick: (e: any) => void
  changeDesktopSettings: (settings: any) => void
  getConnectState: () => number
}

export class Header extends React.Component<IHeaderProps> {
  render (): JSX.Element {
    return (
      <React.Fragment>
        <div className="header">
          <ConnectButton
            handleConnectClick={this.props.handleConnectClick}
            kvmstate={this.props.kvmstate}
          />
          <DesktopSettings
            changeDesktopSettings={this.props.changeDesktopSettings}
            getConnectState={this.props.getConnectState}
          />
        </div>
      </React.Fragment>
    )
  }
}
