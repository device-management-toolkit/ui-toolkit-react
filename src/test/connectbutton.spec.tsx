/*********************************************************************
* Copyright (c) Intel Corporation 2019
* SPDX-License-Identifier: Apache-2.0
**********************************************************************/

import React from 'react'
import { ConnectButton, ConnectProps } from '../reactjs/KVM/ConnectButton'
import { shallow } from 'enzyme'

describe('Test ConnectButton', () => {
  it('Test render() in ConnectButton with kvmstate == 1', () => {
    // Initialization of ConnectProps
    const connectbuttonprops: ConnectProps = {
      kvmstate: (1),
      handleConnectClick: (event: React.MouseEvent) => {}
    }

    const cb = shallow(<ConnectButton {...connectbuttonprops} />)

    // Output
    expect(cb.prop('onClick')).toBe(connectbuttonprops.handleConnectClick)
    expect(cb.prop('children')).toBe('Connecting KVM')
    expect(cb).toMatchSnapshot()
    console.log(cb.debug())
    console.log(cb.props())
  })

  it('Test render() in ConnectButton with kvmstate == 2', () => {
    // Initialization of ConnectProps
    const connectbuttonprops: ConnectProps = {
      kvmstate: (2),
      handleConnectClick: (event: React.MouseEvent) => {}
    }

    const cb = shallow(<ConnectButton {...connectbuttonprops} />)

    // Output
    expect(cb.prop('onClick')).toBe(connectbuttonprops.handleConnectClick)
    expect(cb.prop('children')).toBe('Disconnect KVM')
    expect(cb).toMatchSnapshot()
    console.log(cb.debug())
    console.log(cb.props())
  })

  it('Test render() in ConnectButton with kvmstate == 0', () => {
    // Initialization of ConnectProps
    const connectbuttonprops: ConnectProps = {
      kvmstate: (0),
      handleConnectClick: (event: React.MouseEvent) => {}
    }

    const cb = shallow(<ConnectButton {...connectbuttonprops} />)

    // Output
    expect(cb.prop('onClick')).toBe(connectbuttonprops.handleConnectClick)
    expect(cb.prop('children')).toBe('Connect KVM')
    expect(cb).toMatchSnapshot()
    console.log(cb.debug())
    console.log(cb.props())
  })
})
