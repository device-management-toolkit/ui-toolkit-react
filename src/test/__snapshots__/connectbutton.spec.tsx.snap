// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ConnectButton Test render() in ConnectButton with kvmstate == 0 1`] = `ShallowWrapper {}`;

exports[`Test ConnectButton Test render() in ConnectButton with kvmstate == 1 1`] = `ShallowWrapper {}`;

exports[`Test ConnectButton Test render() in ConnectButton with kvmstate == 2 1`] = `ShallowWrapper {}`;
